src/main/java/uk/gov/ida/verifyserviceprovider/factories/VerifyServiceProviderFactory.java

```diff
@@ -138,10 +139,10 @@ private MetadataCredentialResolver getHubMetadataCredentialResolver() {
     }
 
     private MetadataResolver getMsaMetadataResolver() {
-        return msaMetadataBundle.getMetadataResolver();
+        return msaMetadataBundle.map(item -> item.getMetadataResolver()).get();
     }
 
     private ExplicitKeySignatureTrustEngine getMsaSignatureTrustEngine() {
-        return msaMetadataBundle.getSignatureTrustEngine();
+        return msaMetadataBundle.map(item->item.getSignatureTrustEngine()).get();
     }
```

Comment:
You could use method references instead of the two lambdas above, although it's all of a muchness really.  IntelliJ would prefer us to use method references though.

Reply:
Thanks for this. Will change this